functions/vpn.bash

```diff
@@ -169,3 +169,61 @@ setup_wireguard() {
   echo -n "$(timestamp) [openHABian] Generating QR to load config on the client side (download Wireguard app from PlayStore or AppStore)... "
   qrencode -t ansiutf8 </etc/wireguard/wg0-client.conf
 }
+
+
+## Install Tailscale from their own repo
+## Valid arguments: "install" or "remove"
+##
+##   install_tailscale(String action)
+##
+install_tailscale() {
+  local queryText="We will install the tailscale VPN client on your system. Use it to securely interconnect multiple openHAB(ian) instances.\\nSee https://tailscale.com/blog/how-tailscale-works/ for a comprehensive explanation how it creates a secure VPN. For personal use, you can get a free solo service from tailscale.com."
+  local serviceTargetDir="/lib/systemd/system"
+
+  if [[ -n "$INTERACTIVE" ]]; then
+    if (whiptail --title "VPN setup" --yes-button "Continue" --no-button "Cancel" --yesno "$queryText" 12 80); then echo "OK"; else echo "CANCELED"; return 1; fi
+  fi
+
+  if [[ "$1" == "remove" ]]; then
+    cond_redirect systemctl disable tailscaled.service
+    rm -f ${serviceTargetDir}/tailscale*
+    if ! cond_redirect systemctl -q daemon-reload &> /dev/null; then echo "FAILED (daemon-reload)"; return 1; fi
+    if ! apt-get purge --yes tailscale; then echo "FAILED (purge tailscale)"; return 1; fi
+    return 0
+  fi
+  if [[ "$1" != "install" ]]; then echo "FAILED"; return 1; fi
+
+  # Add Tailscale's GPG key
+  add_keys https://pkgs.tailscale.com/stable/raspbian/buster.gpg
+  # Add the tailscale repository
+  curl https://pkgs.tailscale.com/stable/raspbian/buster.list | tee /etc/apt/sources.list.d/tailscale.list
+  if ! cond_redirect apt-get update; then echo "FAILED (update apt lists)"; return 1; fi
+  # Install Tailscale
+  if cond_redirect apt-get install --yes tailscale; then echo "OK"; else echo "FAILED (install tailscale)"; return 1; fi
+
+  return 0
+}
+
+
+## add node to private Tailscale network
+##
+##   setup_tailscale(String action)
+##
+setup_tailscale() {
+  local preAuthKey=${preauthkey:-preauthkey}
+
+  if [[ -n $UNATTENDED ]] && [[ -z $preAuthKey ]]; then
+    echo "$(timestamp) [openHABian] Beginning tailscale VPN setup... CANCELED (no pre auth key provided)"
+    return 0
+  fi
+  if [[ -n "$INTERACTIVE" ]]; then
+    if ! preAuthKey="$(whiptail --title "Enter pre auth key" --inputbox "\\nIf you have not received / created the Tailscale pre auth key at this stage, please do so now or tell your administrator to. This can be done on the admin console. There's a menu option on the Tailscale Windows client to lead you there.\\n\\nPlease enter the Tailscale pre auth key for this system:" 11 80 "$preAuthKey" 3>&1 1>&2 2>&3)"; then echo "CANCELED"; return 0; fi
+  fi
+
+  if ! tailscale up --authkey "${preAuthKey}"; then echo "FAILED (join Tailscale VPN)"; return 1; fi 
+  # shellcheck disable=SC2154
+  tailscale status | mail -s "openHABian client joined tailscale VPN" "$adminmail"
```

Comment:
Add a check to ensure that exim is installed if interactive.

Reply:
done